### PR TITLE
Jdbc driver is compatible with multiple versions

### PR DESCRIPTION
**Pull request by sjgllgh**
### What is the purpose of the change

The Jdbc driver is compatible with multiple versions

### Related issues/PRs

Related issues: #4929

### Checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [Linkis mailing list](https://linkis.apache.org/community/how-to-subscribe) first)
- [ ] **If this is a code change**: I have written unit tests to fully verify the new behavior.

### CODE REVIEW

**Review by casionone — APPROVED** [review]
LGTM

File: linkis-computation-governance/linkis-jdbc-driver/src/main/scala/org/apache/linkis/ujes/jdbc/LinkisSQLConnection.scala [hunk 1]
```diff
@@ -113,7 +114,17 @@ class LinkisSQLConnection(private[jdbc] val ujesClient: UJESClient, props: Prope
       if (params != null & params.length() > 0) {
         params.split(PARAM_SPLIT).map(_.split(KV_SPLIT)).foreach {
           case Array(k, v) if k.equals(UJESSQLDriver.ENGINE_TYPE) =>
-            return EngineTypeLabelCreator.createEngineTypeLabel(v)
+            if (v.contains('-')) {
+              val factory = LabelBuilderFactoryContext.getLabelBuilderFactory
+              val label = factory.createLabel(classOf[EngineTypeLabel])
+              val engineType = v.split("-").head
```
Comment (peacewong) [inline linkis-computation-governance/linkis-jdbc-driver/src/main/scala/org/apache/linkis/ujes/jdbc/LinkisSQLConnection.scala:120]: should use setStringValue

File: linkis-computation-governance/linkis-jdbc-driver/src/main/scala/org/apache/linkis/ujes/jdbc/LinkisSQLConnection.scala [hunk 2]
```diff
@@ -442,7 +453,7 @@ class LinkisSQLConnection(private[jdbc] val ujesClient: UJESClient, props: Prope
 
   private[jdbc] def toSubmit(code: String): JobExecuteResult = {
     val engineTypeLabel = getEngineType
-    labelMap.put(LabelKeyConstant.ENGINE_TYPE_KEY, engineTypeLabel.getStringValue)
+    labelMap.put(LabelKeyConstant.ENGINE_TYPE_KEY, engineTypeLabel)
```
Comment (peacewong) [inline linkis-computation-governance/linkis-jdbc-driver/src/main/scala/org/apache/linkis/ujes/jdbc/LinkisSQLConnection.scala:456]: should use getStringValue

File: linkis-computation-governance/linkis-jdbc-driver/src/main/scala/org/apache/linkis/ujes/jdbc/LinkisSQLConnection.scala (hunk 1 above)
Comment (sjgllgh) [reply linkis-computation-governance/linkis-jdbc-driver/src/main/scala/org/apache/linkis/ujes/jdbc/LinkisSQLConnection.scala]: please review again

File: linkis-computation-governance/linkis-jdbc-driver/src/main/scala/org/apache/linkis/ujes/jdbc/LinkisSQLConnection.scala (hunk 2 above)
Comment (sjgllgh) [reply linkis-computation-governance/linkis-jdbc-driver/src/main/scala/org/apache/linkis/ujes/jdbc/LinkisSQLConnection.scala]: please review again

**Review by peacewong — APPROVED** [review]
LGTM.

**Pull request merged**